src/bounds.jl

```diff
@@ -115,134 +197,136 @@ function bounds_propagation(m::PODNonlinearModel)
 """
-function resolve_lifted_var_bounds(m::PODNonlinearModel)
+function resolve_var_bounds(m::PODNonlinearModel)
+
+    # Basic Bound propagation
+    m.presolve_bp && bounds_propagation(m) # Fetch bounds from constraints
+
+    # First resolve infinity bounds with assumptions
+    resolve_inf_bounds(m) # Temporarily disabled
 
     # Added sequential bound resolving process base on DFS process, which ensures all bounds are secured.
     # Increased complexity from linear to square but a reasonable amount
     # Potentially, additional mapping can be applied to reduce the complexity
-    # TODO: need to consider negative values
-    for i in 1:length(m.nonlinear_terms)
-        for bi in keys(m.nonlinear_terms)
-            if (m.nonlinear_terms[bi][:id]) == i && (m.nonlinear_terms[bi][:nonlinear_type] in [:bilinear, :monomial, :multilinear])
-                lifted_idx = m.nonlinear_terms[bi][:lifted_var_ref].args[2]
-                cnt = 0
-                bound = []
-                for var in bi
-                    cnt += 1
-                    var_idx = var.args[2]
-                    var_bounds = [m.l_var_tight[var_idx], m.u_var_tight[var_idx]]
-                    if cnt == 1
-                        bound = copy(var_bounds)
-                    elseif cnt == 2
-                        bound = bound * var_bounds'
-                    else
-                        bound = diag(bound) * var_bounds'
-                    end
-                end
-                if minimum(bound) > m.l_var_tight[lifted_idx] + m.tol
-                    m.l_var_tight[lifted_idx] = minimum(bound)
-                end
-                if maximum(bound) < m.u_var_tight[lifted_idx] - m.tol
-                    m.u_var_tight[lifted_idx] = maximum(bound)
-                end
-            end
+    for i in 1:length(m.term_seq)
+        k = m.term_seq[i]
+        if haskey(m.nonconvex_terms, k)
+            m.nonconvex_terms[k][:bound_resolver](m, k)
+        elseif haskey(m.linear_terms, k)
+            basic_linear_bounds(m, k)
+        else
+            error("[RARE] Found homeless term key $(k) during bound resolution.")
         end
     end
 
-    # Resolve bounds for lifted linear terms
-    for i in keys(m.linear_terms)
-        lifted_idx = m.linear_terms[i][:y_idx]
-        ub = 0.0
-        lb = 0.0
-        for j in m.linear_terms[i][:ref][:coef_var]
-            (j[1] > 0.0) ? ub += abs(j[1])*m.u_var_tight[j[2]] : ub -= abs(j[1])*m.l_var_tight[j[2]]
-            (j[1] > 0.0) ? lb += abs(j[1])*m.l_var_tight[j[2]] : lb -= abs(j[1])*m.u_var_tight[j[2]]
+    return
+end
+
+"""
+    Critically assumed since POD relies on finite bound to work
+"""
+function resolve_inf_bounds(m::PODNonlinearModel)
+
+    warnuser = false
+    infcount = 0
+
+    # Only specify necessary bounds
+    for i in m.candidate_disc_vars
+        if m.l_var_tight[i] == -Inf
+            warnuser = true
```

Comment:
Here is -10e4 a good enough M value - do we want to make this problem specific

Reply:
The short answer is NO as the problem can be specific. I have this as a parameter in the next PR.